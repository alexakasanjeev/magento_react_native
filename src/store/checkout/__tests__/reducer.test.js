import checkoutReducer, { INITIAL_STATE } from '../reducer';
import {
  MAGENTO,
  RESET_PAYMENT_STATE,
  RESET_CHECKOUT_STATE,
  RESET_SHIPPING_STATE,
  RESET_CHECKOUT_ADDRESS_STATE,
} from '../../../constants';
import Status from '../../../magento/Status';

describe('checkout reducer', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = checkoutReducer(undefined, {});

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart billing address loading', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_BILLING_ADDRESS_LOADING,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart billing address success', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_BILLING_ADDRESS_SUCCESS,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart billing address failure', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_BILLING_ADDRESS_FAILURE,
      payload: {
        errorMessage: 'Unable to add billing address to cart.',
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle shipping method loading', () => {
    // Setup
    const action = {
      type: MAGENTO.GET_SHIPPING_METHOD_LOADING,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle shipping method success', () => {
    // Setup
    const action = {
      type: MAGENTO.GET_SHIPPING_METHOD_SUCCESS,
      payload: {
        shipping: [
          {
            base_amount: 5,
            carrier_title: 'Mock Shipping title',
            method_title: 'MST',
            carrier_code: 'MST',
          }
        ]
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle shipping method failure', () => {
    // Setup
    const action = {
      type: MAGENTO.GET_SHIPPING_METHOD_FAILURE,
      payload: {
        errorMessage: 'No shipping method available.',
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart shipping info loading', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_SHIPPING_INFO_LOADING,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart shipping info success', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_SHIPPING_INFO_SUCCESS,
      payload: {
        payment: {
          totals: {
            // data related to price
          },
          payment_methods: [
            {
              code: 'COD',
              title: 'Cash On Delivery',
            }
          ]
        }
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle add cart shipping info failure', () => {
    // Setup
    const action = {
      type: MAGENTO.ADD_CART_SHIPPING_INFO_FAILURE,
      payload: {
        errorMessage: 'Unable to add cart shipping info.',
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle place cart order loading', () => {
    // Setup
    const action = {
      type: MAGENTO.PLACE_CART_ORDER_LOADING,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle place cart order success', () => {
    // Setup
    const action = {
      type: MAGENTO.PLACE_CART_ORDER_SUCCESS,
      payload: {
        orderId: 23
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle place cart order failure', () => {
    // Setup
    const action = {
      type: MAGENTO.PLACE_CART_ORDER_FAILURE,
      payload: {
        errorMessage: 'Unable to place order, something went wrong.',
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle order detail loading', () => {
    // Setup
    const action = {
      type: MAGENTO.ORDER_DETAIL_LOADING,
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle order detail success', () => {
    // Setup
    const action = {
      type: MAGENTO.ORDER_DETAIL_SUCCESS,
      payload: {
        order: {
          // dummy order object, not actual representation
          orderId: 23
        }
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle order detail failure', () => {
    // Setup
    const action = {
      type: MAGENTO.ORDER_DETAIL_FAILURE,
      payload: {
        errorMessage: 'Unable to get order detail.',
      }
    };

    // Exercise
    const newState = checkoutReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset payment state', () => {
    // Setup
    const action = {
      type: RESET_PAYMENT_STATE,
    };
    const initialState = {
      ...INITIAL_STATE,
      paymentMethodStatus: Status.SUCCESS,
      payment: {
        totals: {},
        payment_methods: [
          {
            code: 'COD',
            title: 'Cash On Delivery',
          }
        ]
      },
    };

    // Exercise
    const newState = checkoutReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset shipping state', () => {
    // Setup
    const action = {
      type: RESET_SHIPPING_STATE,
    };
    const initialState = {
      ...INITIAL_STATE,
      shippingMethodStatus: Status.SUCCESS,
      billingAddressStatus: Status.SUCCESS,
      shipping: [
        {
          base_amount: 5,
          carrier_title: 'Mock Shipping title',
          method_title: 'MST',
          carrier_code: 'MST',
        }
      ],
    };

    // Exercise
    const newState = checkoutReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset checkout address state', () => {
    // Setup
    const action = {
      type: RESET_CHECKOUT_ADDRESS_STATE,
    };
    const initialState = {
      ...INITIAL_STATE,
      billingAddressStatus: Status.SUCCESS
    };

    // Exercise
    const newState = checkoutReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle reset entire checkout state', () => {
    // Setup
    const action = {
      type: RESET_CHECKOUT_STATE,
    };
    const initialState = {
      ...INITIAL_STATE,
      shippingMethodStatus: Status.SUCCESS,
      billingAddressStatus: Status.SUCCESS,
      shipping: [
        {
          base_amount: 5,
          carrier_title: 'Mock Shipping title',
          method_title: 'MST',
          carrier_code: 'MST',
        }
      ],
      paymentMethodStatus: Status.SUCCESS,
      payment: {
        totals: {},
        payment_methods: [
          {
            code: 'COD',
            title: 'Cash On Delivery',
          }
        ]
      },
    };

    // Exercise
    const newState = checkoutReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });
});
