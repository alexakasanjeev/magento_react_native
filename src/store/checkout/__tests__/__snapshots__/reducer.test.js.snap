// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout reducer should handle add cart billing address failure 1`] = `
Object {
  "billingAddressStatus": "ERROR",
  "errorMessage": "Unable to add billing address to cart.",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle add cart billing address loading 1`] = `
Object {
  "billingAddressStatus": "LOADING",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle add cart billing address success 1`] = `
Object {
  "billingAddressStatus": "SUCCESS",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle add cart shipping info failure 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "Unable to add cart shipping info.",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "ERROR",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle add cart shipping info loading 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "LOADING",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle add cart shipping info success 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": Object {
    "payment_methods": Array [
      Object {
        "code": "COD",
        "title": "Cash On Delivery",
      },
    ],
    "totals": Object {},
  },
  "paymentMethodStatus": "SUCCESS",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle order detail failure 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "Unable to get order detail.",
  "orderDetailStatus": "ERROR",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle order detail loading 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "LOADING",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle order detail success 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "order": Object {
    "orderId": 23,
  },
  "orderDetailStatus": "SUCCESS",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle place cart order failure 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "Unable to place order, something went wrong.",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "ERROR",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle place cart order loading 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "LOADING",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle place cart order success 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderId": 23,
  "orderStatus": "SUCCESS",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle reset checkout address state 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle reset entire checkout state 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle reset payment state 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle reset shipping state 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle shipping method failure 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "No shipping method available.",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "ERROR",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle shipping method loading 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "LOADING",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should handle shipping method success 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": Array [
    Object {
      "base_amount": 5,
      "carrier_code": "MST",
      "carrier_title": "Mock Shipping title",
      "method_title": "MST",
    },
  ],
  "shippingMethodStatus": "SUCCESS",
  "status": "NOT_STARTED",
}
`;

exports[`checkout reducer should return the initial state 1`] = `
Object {
  "billingAddressStatus": "NOT_STARTED",
  "errorMessage": "",
  "orderDetailStatus": "NOT_STARTED",
  "orderStatus": "NOT_STARTED",
  "payment": null,
  "paymentMethodStatus": "NOT_STARTED",
  "shipping": null,
  "shippingMethodStatus": "NOT_STARTED",
  "status": "NOT_STARTED",
}
`;
